### ISSUE TITLE
Reproduce Bao et al on CDCP

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-document-level#368]
This is the command that should reproduce the setting from [Bao et al](https://aclanthology.org/2022.emnlp-main.713.pdf):
```
python src/train.py \
experiment=generic_pointer_network_end2end_re_past \
dataset=cdcp_prepared \
dataset.create_validation_split.seed=12345 \
dataset.create_validation_split.test_size=0.15 \
model.base_model_config.lr=8e-5 \
trainer=gpu \
trainer.min_epochs=75 \
trainer.max_epochs=75 \
trainer.deterministic=true \
test=true \
name=dataset-cdcp/task-ner_re/bao \
seed=1,12,123,1234,12345 \
--multirun
```

Notes (differences to Bao et al in capital):
 - `seed=12345`: use a fixed seed for validation set creation. THIS WILL RESULT IN A DIFFERENT VAL SET THAN BAO ET AL USED! THEY ALSO CREATE THE VAL SET ON THE PARAGRAPH LEVEL, WE DO IT ON THE DOCUMENT LEVEL.
 - `test_size=0.15`: set val size to 15% (of train)
 - `lr=8e-5`: same learning rate as Bao et al
 - `(min|max)_epochs=75`: train for 75 epochs
 - `deterministic=true`: to make this again reproducible
 - train 5 runs with different. BAO TRAINED 10 RUNS.
 - we implicitly use `monitor_metric=metric/binary_relations/micro/f1/val`. BAO USES `(metric/binary_relations/micro/f1/val + metric/labeled_spans/micro/f1/val) / 2`!
 - we use `batch_size=2` (implicitly) to run on RTX2080Ti. BAO USES `batch_size=32`
 - BAO USES A BucketSampler

**Comment by ArneBinder** [comment]
This can be found here: https://github.com/ArneBinder/pie-document-level/pull/365#issuecomment-2643391689
